zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -154,6 +176,19 @@ public static CompletableFuture<ResponseEntity<Object>> executeServiceMethod(
                     .orElseGet(result));
   }
 
+  public static <BrokerResponseT, RestResponseT>
+      CompletableFuture<ResponseEntity<RestResponseT>> executeServiceMethod(
+          final Supplier<CompletableFuture<BrokerResponseT>> method,
+          final Function<BrokerResponseT, ResponseEntity<RestResponseT>> result) {
```

Comment:
💭 I haven't tried but at a first glance I would hope we can merge this method with the `executeServiceMethod` above or at least reuse one from the other.